Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,143 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'EN' : 'DIS';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const fanEnableTextColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
+
+    const chargeEnableText = (props.data?.charge_enable[0] ?? -1.0) ? 'Enable' : 'Disable';
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const chargeEnableTextColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
+
+    const fanSpeedMin = ((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
+                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
+                         CONSTANTS.fan_speed.MAX;
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
+                <Flex w='95%'>
+                    <Center
+                        border='2px'
+                        borderRadius='0px'
+                        h='75%'
+                        background={fanEnableColor}
+                        marginTop='1%'
+                        marginBottom='1%'
```

Comment:
```suggestion
                    <Center
                        border='2px'
                        w='2em'
                        background={fanEnableColor}
                        py='3px'
```
A few things:
- No need to specify a borderRadius of 0px
- For the width, `2em` worked for me, but the big idea is that it would be nice to set the width of the EN/DIS Center so that there is some horizontal space between the text and borders and the width doesn't change when the text switches. I wouldn't decrease the size more than that, but it might need to be increased to work for you
- The height can be removed since the Center will naturally shrink to fit its contents in this case (see later comment on Text for adjusting height)
- Can use py (vertical padding) instead of a vertical margin on the text to add space between the text and the borders
- No need for the top/bottom margins since the Center will be vertically centered within the Flex and will shrink to fit the Text + padding

Reply:
Removing the top and bottom margins makes it hit the top and bottom of the container on my computer, are you ok with that? 
<img width="60" alt="image" src="https://github.com/badgerloop-software/chase-car-dashboard/assets/17338790/8db240ea-e653-4c46-801a-f7f9434e5501">